[Datasets] Remove "ML Preprocessing" user guide (#32965)
ray-project/ray: doc/source/data/dataset.rst

Data Loading and Preprocessing for ML Training
----------------------------------------------

Use Ray Datasets to load and preprocess data for distributed :ref:`ML training pipelines <train-docs>`.
Compared to other loading solutions, Datasets are more flexible (e.g., can express higher-quality `per-epoch global shuffles <examples/big_data_ingestion.html>`__) and provides `higher overall performance <https://www.anyscale.com/blog/why-third-generation-ml-platforms-are-more-performant>`__.

> Use Datasets for the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray. Do not use them as a replacement for more general data processing systems.

Reviewer: ```suggestion
Use Datasets as a last-mile bridge from storage or ETL pipeline outputs to distributed applications and libraries in Ray. Do not use it as a replacement for more general data processing systems.
```

Its utility is as the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray:

.. image:: images/dataset-loading-1.png
   :width: 650px
   :align: center

..
  https://docs.google.com/presentation/d/1l03C1-4jsujvEFZUM4JVNy8Ju8jnY5Lc_3q7MBWi2PQ/edit

You can also use Datasets seamlessly with Ray-integrated DataFrame libraries.

.. image:: images/dataset-loading-2.png
   :width: 650px

>    :align: center

Reviewer 2: I am thinking of to drop this part. If we look at the images, they are mostly identical, except the scope of "Ray cluster". And to run Spark/Mars/etc on Ray is community-maintained, and not battle tested. So it should be safe to drop here.

Author: Removed

> Ray Datasets are designed to load and preprocess data for distributed :ref:`ML training pipelines <train-docs>`.

Reviewer 3: ```suggestion
Use Ray Datasets to load and preprocess data for distributed :ref:`ML training pipelines <train-docs>`.
```

Compared to other loading solutions, Datasets are more flexible (e.g., can express higher-quality `per-epoch global shuffles <examples/big_data_ingestion.html>`__) and provides `higher overall performance <https://www.anyscale.com/blog/why-third-generation-ml-platforms-are-more-performant>`__.

> Datasets is not intended as a replacement for more general data processing systems.

Reviewer 3: ```suggestion
Use Datasets for the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray. Do not use them as a replacement for more general data processing systems.
```

> Its utility is as the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray:

Reviewer 3: ```suggestion

```

Reviewer 3: If you're comfortable being more opinionated about this guidance. Think of how you would tell a friend who is using Datasets.

Also, do we need to explain what we mean by `last-mile`? I thought that might be a term that we coined internally. If not, no need to explain.

Author: Hmm...really not sure how I'd describe Datasets' utility in a simple but precise way.

Maybe something like "Datasets moves data from storage to distributed Ray applications. It also lets you transform data into representations that are more suitable for models."?

Reviewer 2: Can we remove "Its utility is as the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray:".

This is pretty much repeated from above sentence: "Use Datasets for the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray."

We can add a link to "User Guides" here, e.g. "Check out Datasets User Duides for the features it supports".

.. image:: images/dataset-loading-1.png
   :width: 650px
   :align: center

..
  https://docs.google.com/presentation/d/1l03C1-4jsujvEFZUM4JVNy8Ju8jnY5Lc_3q7MBWi2PQ/edit

> Ray-integrated DataFrame libraries can also be seamlessly used with Datasets, to enable running a full data to

Reviewer 3: ```suggestion
You can also use Datasets seamlessly with Ray-integrated DataFrame libraries. You can run a data to
```

> ML pipeline completely within Ray without requiring data to be materialized to external storage:

Reviewer 3: ```suggestion
ML pipeline completely within Ray without materializing data to external storage:
```

Reviewer 3: Not sure if I mangled the intended meaning, but I'm trying to avoid some passive voice.

> ML pipeline completely within Ray without requiring data to be materialized to external storage:

Reviewer 3: ```suggestion
ML pipeline completely within Ray without requiring data to be materialized to external storage.
```
